<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>851</width>
    <height>433</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Лабораторная работа №1-2: Сортировки</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTextBrowser" name="textArrayDisplay">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>661</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelArrayDisplay">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Массив</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonInsertionSort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Сортировка простыми вставками</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonSelectionSort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Сортировка простым выбором</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonBubbleSort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Сортировка простым обменом</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonShellSort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Сортировка Шелла</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonTetragonSort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Сортировка квадратичной выборки</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonQuickSort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Быстрая сортировка</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonMergeSort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Сортировка слиянием</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDisplayTimeInsertionSort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>80</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDisplayTimeSelectionSort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>120</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDisplayTimeBubbleSort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>160</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDisplayTimeShellSort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>200</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDisplayTimeTetragonSort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>240</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDisplayTimeQuickSort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>280</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDisplayTimeMergeSort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>320</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textArrayLengthDisplay">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>30</y>
      <width>161</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelArrayLengthDisplay">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>10</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Длина массива</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>851</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="actionAppend"/>
    <addaction name="actionGenerateRandomArray"/>
    <addaction name="actionRemove"/>
    <addaction name="actionDelete"/>
    <addaction name="actionSave"/>
    <addaction name="actionUndo"/>
    <addaction name="separator"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionAppend"/>
   <addaction name="actionGenerateRandomArray"/>
   <addaction name="actionRemove"/>
   <addaction name="actionDelete"/>
   <addaction name="actionSave"/>
   <addaction name="actionUndo"/>
   <addaction name="separator"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSaveAs"/>
   <addaction name="separator"/>
   <addaction name="actionSettings"/>
   <addaction name="actionExit"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionAppend">
   <property name="icon">
    <iconset>
     <normaloff>icons/add.png</normaloff>icons/add.png</iconset>
   </property>
   <property name="text">
    <string>Добавить элемент массива</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionRemove">
   <property name="icon">
    <iconset>
     <normaloff>icons/pop.png</normaloff>icons/pop.png</iconset>
   </property>
   <property name="text">
    <string>Удалить последний элемент массива</string>
   </property>
   <property name="iconText">
    <string>Удалить последний элемент массива</string>
   </property>
   <property name="toolTip">
    <string>Удалить последний элемент массива</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset>
     <normaloff>icons/delete.png</normaloff>icons/delete.png</iconset>
   </property>
   <property name="text">
    <string>Удалить массив</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+D</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset>
     <normaloff>icons/undo.png</normaloff>icons/undo.png</iconset>
   </property>
   <property name="text">
    <string>Загрузить массив из памяти</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset>
     <normaloff>icons/exit.png</normaloff>icons/exit.png</iconset>
   </property>
   <property name="text">
    <string>Закрыть</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionGenerateRandomArray">
   <property name="icon">
    <iconset>
     <normaloff>icons/random.png</normaloff>icons/random.png</iconset>
   </property>
   <property name="text">
    <string>Сгенерировать случайный массив</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset>
     <normaloff>icons/saveas.png</normaloff>icons/saveas.png</iconset>
   </property>
   <property name="text">
    <string>Сохранить как...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>icons/open.png</normaloff>icons/open.png</iconset>
   </property>
   <property name="text">
    <string>Открыть</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>icons/save.png</normaloff>icons/save.png</iconset>
   </property>
   <property name="text">
    <string>Сохранить массив в память</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset>
     <normaloff>icons/settings.png</normaloff>icons/settings.png</iconset>
   </property>
   <property name="text">
    <string>Настройки</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
